src/network/network_server.cpp

```diff
@@ -1643,37 +1644,37 @@ static void NetworkAutoCleanCompanies()
 	}
 
 	/* Go through all the companies */
-	for (const Company *c : Company::Iterate()) {
+	for (Company *c : Company::Iterate()) {
 		/* Skip the non-active once */
 		if (c->is_ai) continue;
 
 		if (!HasBit(has_clients, c->index)) {
 			/* The company is empty for one month more */
-			_network_company_states[c->index].months_empty++;
+			if (c->months_empty != std::numeric_limits<decltype(c->months_empty)>::max()) c->months_empty++;
```

Comment:
`std::numeric_limits` is fine, but I wonder if it would look cleaner if it compared against a new `AUTOCLEAN_MAX_MONTHS` constant that would take the place of the hardcoded 240 for the three autoclean settings in `network_settings.ini`. Probably a potato/potato question. 😄

Reply:
Lets assume `autoclean_novehicles` is `AUTOCLEAN_MAX_MONTHS` and we limit `months_empty` to `AUTOCLEAN_MAX_MONTHS`.
`months_empty > autoclean_novehicles` => `AUTOCLEAN_MAX_MONTHS < AUTOCLEAN_MAX_MONTHS` => false, so auto clean will be disabled.
So the threshold would need to become `AUTOCLEAN_MAX_MONTHS + 1` and then it starts to become doubtful whether it's really better.

Reply:
Ah, yes. 🙂